### DISCUSSION TITLE
I created a space invader clone for the Picocomputer

### DISCUSSION THREAD

**Discussion by marklinebaugh** [picocomputer/community#125] [Q&A]
As my first Picocomputer project I decided to create a clone of the classic Space Invaders video game. Here's a short clip of the game's "demo" mode. https://youtu.be/y0Ij1Ej1WHY?feature=shared

I'm more of a hardware guy, not a programmer, so I wanted (actually "I  needed") to start with something simple. And, despite its simplicity, I have to admit it has been a challenging project. Without the infrastructure/OS provided by Rumbledethumps and other contributors, it would have been a bridge too far for me. 

Kudos to Rumbledethumps and those that have supported this project. It's awe inspiring.

**Comment by rumbledethumps** [comment]
First game with sound! Looking forward to playing it.

**Comment by marklinebaugh** [comment]
Thanks! I'll make it available, soon... after I learn a bit more about github

**Comment by marklinebaugh** [comment]
I think I've got it. Comments and feedback are very much appreciated.   

https://github.com/marklinebaugh/RP6502-Space-Raiders

**Comment by rumbledethumps** [comment]
I see you paid a lot of attention to the details. All three bomb types. The last alien moves faster when moving right. An attract mode. This is awesome. The little details are what made games like this survive the test of time.

Only two requests. I bring Tony's (@vruumllc) fantastic battery-powered all-in-one to various events and just leave it for folks to play with. Usually with a sticker that says something like "press 1 to start". But the start buttons (1 and 2) don't work on every attract mode scene so I'm not sure what to put there. I ended up pressing ctrl-alt-del, typing reset, then quickly hitting 1 before the attract mode changed scenes. It'd be great if 1 and 2 worked all the time so I don't need to talk someone through starting a game. The other request is to change the score filename to "raiders.hiscore".

**Reply by marklinebaugh** [reply]
Thanks for the kind words. At one point, I had the functionality you've requested. I must have inadvertently lost it along the way. I just published an update, which (I think) provides what you've asked for. If not, please let me know.

https://github.com/marklinebaugh/RP6502-Space-Raiders

**Reply by rumbledethumps** [reply]
Much better. It doesn't work on the demo game over scene and doesn't redraw the score, but I can leave it with a sticker now. Thanks.

**Reply by marklinebaugh** [reply]
I believe I've addressed all the short "keyboard input, black out periods" which impacted the ability to immediately transition from demo mode to play mode. Also, I changed the default "score" display to include player 2's score (to match the demo mode). After initialization, the player 2 score is displayed unless and until single player mode is selected.

**Comment by GlennFR** [comment]
Well done and thank-you!  Very impressive results.  And now I remember why the sound on/off feature was so important on PC games :)
Seriously, this is an awesome feat.  And well done to Rumble, also, for this really does prove the universality of the RP6502 design - just download, build and play!

**Comment by marklinebaugh** [comment]
That's very kind, thank you! Yes volume control becomes essential after a couple rounds :) 
And, I agree... Rumble's creativity and ingenuity have produced amazing results! I'm super-impressed!

**Comment by James-HoneyBadger** [comment]
Just downloaded & compiled the game - impressive job!  Love the sound effects :)

**Reply by marklinebaugh** [reply]
Thanks! Creation of the sound effects was one of the most fun and interesting parts of the design. Glad you liked them.

**Reply by cskordis** [reply]
For someone like me who doesn't know how to compile, is the a way on running the code? is there executibles somewhere

**Reply by jasonfrowe** [reply]
> For someone like me who doesn't know how to compile, is the a way on running the code? is there executibles somewhere

I suggest starting with:
https://github.com/picocomputer/vscode-cc65
If you can compile the "Hello, World!" example, then you'll be ready to compile other projects as well.
